python/ray/actor.py

```diff
@@ -93,73 +100,123 @@ def get_actor_checkpoint(worker, actor_id):
-    updated to match the executed task. Else, the executor reports failure to
-    the local scheduler so that the task counter does not get updated.
+    The wrapped method updates the worker's internal state and performs any
+    necessary checkpointing operations.
 
     Args:
         worker (Worker): The worker that is executing the actor.
         method_name (str): The name of the actor method.
         method (instancemethod): The actor method to wrap. This should be a
             method defined on the actor class and should therefore take an
             instance of the actor as the first argument.
+        actor_imported (bool): Whether the actor has been imported.
+            Checkpointing operations will not be run if this is set to False.
 
     Returns:
         A function that executes the given actor method on the worker's stored
             instance of the actor. The function also updates the worker's
             internal state to record the executed method.
     """
 
-    def actor_method_executor(dummy_return_id, task_counter, actor,
-                              *args):
-        if method_name == "__ray_checkpoint__":
-            # Execute the checkpoint task.
-            actor_checkpoint_failed, error = method(actor, *args)
-            # If the checkpoint was successfully loaded, update the actor's
-            # task counter and set a flag to notify the local scheduler, so
-            # that the task following the checkpoint can run.
-            if not actor_checkpoint_failed:
-                worker.actor_task_counter = task_counter + 1
-                # Once the actor has resumed from a checkpoint, it counts as
-                # loaded.
-                worker.actor_loaded = True
-            # Report to the local scheduler whether this task succeeded in
-            # loading the checkpoint.
-            worker.actor_checkpoint_failed = actor_checkpoint_failed
-            # If there was an exception during the checkpoint method, re-raise
-            # it after updating the actor's internal state.
-            if error is not None:
-                raise error
-            return None
-        else:
-            # Update the worker's internal state before executing the method in
-            # case the method throws an exception.
-            worker.actor_task_counter = task_counter + 1
-            # Once the actor executes a task, it counts as loaded.
-            worker.actor_loaded = True
-            # Execute the actor method.
-            return method(actor, *args)
+    def actor_method_executor(dummy_return_id, actor, *args):
+        # Update the actor's task counter to reflect the task we're about to
+        # execute.
+        worker.actor_task_counter += 1
+
+        # If this is the first task to execute on the actor, try to resume from
+        # a checkpoint.
+        if actor_imported and worker.actor_task_counter == 1:
```

Comment:
Will this be something that we can do during the "actor creation task" once we make actor creation a task?

Reply:
Hmm that depends on what the actor creation task does...if it imports the actor class, then yes.